templates/react-native-template-realm-js/template/app/AppWrapper.js

```diff
@@ -0,0 +1,181 @@
+  // startups, the logged in user will persist.
+  const [user, setUser] = useState(app.currentUser);
+
+  const [authState, setAuthState] = useState(AuthState.None);
+  const [authVisible, setAuthVisible] = useState(false);
+
+  // If the user presses "login", show the auth screen
+  const handleShowAuth = () => {
+    setAuthVisible(true);
+  };
+
+  // If the user presses "login" from the auth screen, try to log them in
+  // with the supplied credentials
+  const handleLogin = async (email, password) => {
+    setAuthState(AuthState.Loading);
+    const credentials = Realm.Credentials.emailPassword(email, password);
+
+    try {
+      setUser(await app.logIn(credentials));
+      setAuthVisible(false);
+      setAuthState(AuthState.None);
+    } catch (e) {
+      console.log('Error logging in', e);
+      setAuthState(AuthState.LoginError);
+    }
+  };
+
+  // If the user presses "register" from the auth screen, try to register a
+  // new account with the  supplied credentials and login as the newly created user
+  const handleRegister = async (email, password) => {
+    setAuthState(AuthState.Loading);
+
+    try {
+      // Register the user...
+      await app.emailPasswordAuth.registerUser({email, password});
+      // ...then login with the newly created user
+      const credentials = Realm.Credentials.emailPassword(email, password);
+
+      setUser(await app.logIn(credentials));
+      setAuthVisible(false);
+      setAuthState(AuthState.None);
+    } catch (e) {
+      console.log('Error registering', e);
+      setAuthState(AuthState.RegisterError);
+    }
+  };
+
+  // If the user presses "logout", unset the user in state and log the user out
+  // of the Realm app
+  const handleLogout = () => {
+    setUser(null);
+    app.currentUser?.logOut();
+  };
+
+  // If anonymous auth is enabled and no user is logged in, log in as an anonymous user
+  useEffect(() => {
+    if (user || !SYNC_CONFIG.anonymousAuthEnabled) {
+      return;
+    }
+
+    (async () => {
+      const credentials = Realm.Credentials.anonymous();
+      try {
+        setUser(await app.logIn(credentials));
+      } catch (e) {
+        console.log('Error logging in anonymous user', e);
+        throw e;
+      }
+    })();
+  }, [user, app]);
+
+  // Return null if we are waiting for anonymous login to complete
+  if ((!user || !app.currentUser) && SYNC_CONFIG.anonymousAuthEnabled) {
+    return null;
+  }
+
+  // If we are not logged in, or the user has pressed "Login" as an anonymous user,
+  // show the login screen
+  if (authVisible || !user || !app.currentUser) {
+    return (
+      <LoginScreen
+        onLogin={handleLogin}
+        onRegister={handleRegister}
+        authState={authState}
+      />
+    );
+  }
+
+  // If we are logged in, add the sync configuration the the RealmProvider and render the app
+  return (
+    <RealmProvider sync={{user, partitionValue: app.currentUser.id}}>
```

Comment:
I was getting a very cryptic crash `Illegal Realm Path`, because i set the partition key type to `objectId`.  Maybe we need to document this possible issue. (Would be good if we had a backend template that worked with this automatically)